server/oauth2.go

```diff
@@ -453,6 +454,12 @@ func (s *Server) parseAuthorizationRequest(r *http.Request) (*storage.AuthReques
 		return &authErr{state, redirectURI, typ, fmt.Sprintf(format, a...)}
 	}
 
+	// dex doesn't support request parameter and must return request_not_supported error
+	// https://openid.net/specs/openid-connect-core-1_0.html#6.1
+	if q.Get("request") != "" {
+		return nil, newErr(errRequestNotSupported, "Server does not support request parameter.")
```

Comment:
Do I recall correctly that the first implementation involved some redirections? I briefly looked at that version and I didn't really understand it.

Reply:
I realized that this Dex already does the necessary redirect if an error is returned from the `parseAuthorizationRequest` method.